src/txn/txn_rollback_to_stable.c

```diff
@@ -110,20 +45,200 @@ __txn_abort_newer_update(
             upd->durable_ts = upd->start_ts = WT_TS_NONE;
         }
     }
+
+    /* Reset the history store flag for the stable update. */
+    if (first_upd != NULL)
+        F_CLR(first_upd, WT_UPDATE_HS);
+
+    return (first_upd);
 }
 
 /*
  * __txn_abort_newer_insert --
- *     Apply the update abort check to each entry in an insert skip list
+ *     Apply the update abort check to each entry in an insert skip list and return the first
+ *     satisfy update.
```

Comment:
I do not understand what `return the first satisfy update.` means. Can you please reword that?

Reply:
I Changed the approach and also the comment.